src/tigerbeetle/repl.zig

```diff
@@ -0,0 +1,813 @@
+                        if (comptime (!std.mem.eql(u8, field.name, "flags") and
+                            !std.mem.eql(u8, field.name, "reserved") and
+                            !std.mem.eql(u8, field.name, "timestamp")))
+                        {
+                            @field(
+                                @field(out.*, enum_field.name),
+                                field.name,
+                            ) = try std.fmt.parseInt(
+                                field.field_type,
+                                value,
+                                10,
+                            );
+                        }
+
+                        // Handle flags, specific to Account and Transfer fields.
+                        if (comptime std.mem.eql(u8, field.name, "flags") and
+                            !std.mem.eql(u8, enum_field.name, "id"))
+                        {
+                            var flags = std.mem.split(u8, value, "|");
+                            var f = std.mem.zeroInit(field.field_type, .{});
+                            while (flags.next()) |flag| {
+                                var flag_no_ws = std.mem.trim(
+                                    u8,
+                                    flag,
+                                    std.ascii.spaces[0..],
+                                );
+                                inline for (@typeInfo(
+                                    field.field_type,
+                                ).Struct.fields) |flag_field| {
+                                    if (std.mem.eql(u8, flag_field.name, flag_no_ws)) {
+                                        if (comptime !std.mem.eql(
+                                            u8,
+                                            flag_field.name,
+                                            "padding",
+                                        )) {
+                                            @field(f, flag_field.name) = true;
+                                        }
+                                    }
+                                }
+                            }
+                            @field(@field(out.*, enum_field.name), "flags") = f;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    // Statement grammar parsed here.
+    // STATEMENT: OPERATION ARGS [;]
+    // OPERATION: create_accounts | lookup_accounts | create_transfers | lookup_transfers
+    //      ARGS: ARG [, ARG]
+    //       ARG: KEY = VALUE
+    //       KEY: string
+    //     VALUE: string [| VALUE]
+    //
+    // For example:
+    //   create_accounts id=1 code=2 ledger=3, id = 2 code= 2 ledger =3;
+    //   create_accounts flags=linked | debits_must_not_exceed_credits ;
+    pub fn parse_statement(
+        arena: *std.heap.ArenaAllocator,
+        input: []const u8,
+    ) !StatementSyntaxTree {
+        var args = std.ArrayList(*ObjectSyntaxTree).init(arena.allocator());
```

Comment:
Hmm I don't understand how that would work here since this is in a separate struct. Within the constraints of being in a separate struct it is as close to where it is used as is possible?

Reply:
> Within the constraints of being in a separate struct it is as close to where it is used as is possible?

I don't understand what you mean here. I'm just suggesting moving this `var args=  ...` line down to just before the `while (parse.offset < parse.input.len) {` loop.

Reply:
Ooops, a different view showed this comment on a different line, or I misread it. I see it correctly now.